test/broker_test/broker_test.go

```diff
@@ -1806,6 +1882,46 @@ func blueprint(setNullFieldsValues bool) func(ctx *TestContext, auth *SMExpect,
 			Ready: true,
 		}).Raw()
 
+		brokerID := obj["id"].(string)
+
+		offerings, err := ctx.SMRepository.List(context.Background(), types.ServiceOfferingType, query.ByField(query.EqualsOperator, "broker_id", brokerID))
+		if err != nil {
+			panic(err)
+		}
+		offeringIDs := make([]string, 0, offerings.Len())
+		for i := 0; i < offerings.Len(); i++ {
+			offeringIDs = append(offeringIDs, offerings.ItemAt(i).GetID())
+		}
+		plans, err := ctx.SMRepository.List(context.Background(), types.ServicePlanType, query.ByField(query.InOperator, "service_offering_id", offeringIDs...))
+		if err != nil {
+			panic(err)
+		}
+		planIDs := make([]string, 0, plans.Len())
+		for i := 0; i < plans.Len(); i++ {
+			planIDs = append(planIDs, plans.ItemAt(i).GetID())
+		}
+		visibilities, err := ctx.SMRepository.List(context.Background(), types.VisibilityType, query.ByField(query.InOperator, "service_plan_id", planIDs...))
+		if err != nil {
+			panic(err)
+		}
+
+		transitiveResourcesExpectedCount := offerings.Len() + plans.Len() + visibilities.Len()
+		transitiveResources := auth.GET(web.ServiceBrokersURL+"/"+brokerID).WithQuery(web.QueryParamLastOp, "true").
+			Expect().Status(http.StatusOK).
+			JSON().Object().Value("last_operation").Object().Value("transitive_resources").Array()
+
+		transitiveResourcesActualCount := 0
+		for _, tr := range transitiveResources.Iter() {
+			// Do not count the notifications and resources which are not for created
+			if tr.Object().Value("type").String().Raw() != types.NotificationType.String() &&
+				tr.Object().Value("operation_type").String().Raw() == string(types.CREATE) {
+				transitiveResourcesActualCount++
+			}
+		}
+		if transitiveResourcesActualCount != transitiveResourcesExpectedCount {
+			Fail(fmt.Sprintf("Expected transitive resources %d but got %d", transitiveResourcesExpectedCount, transitiveResourcesActualCount))
+		}
+
```

Comment:
This logic seems like a test case to me, not part of the resource creation blueprint.

Reply:
Sure, I will try to make it as a test case